pkg/webhook/server/certificate/manager_test.go

```diff
@@ -1,184 +1,185 @@
+			shouldRotate: false,
+		}),
+		Entry("half way expired, still good", setRotationDeadlineCase{
+			notBefore:    -24 * time.Hour,
+			notAfter:     24 * time.Hour,
+			shouldRotate: false,
+		}),
+		Entry("mostly expired, still good", setRotationDeadlineCase{
+			notBefore:    -69 * time.Hour,
+			notAfter:     31 * time.Hour,
+			shouldRotate: false,
+		}),
+		Entry("just about expired, should rotate", setRotationDeadlineCase{
+			notBefore:    -91 * time.Hour,
+			notAfter:     9 * time.Hour,
+			shouldRotate: true,
+		}),
+		Entry("nearly expired, should rotate", setRotationDeadlineCase{
+			notBefore:    -99 * time.Hour,
+			notAfter:     1 * time.Hour,
+			shouldRotate: true,
+		}),
+		Entry("already expired, should rotate", setRotationDeadlineCase{
+			notBefore:    -10 * time.Hour,
+			notAfter:     -1 * time.Hour,
+			shouldRotate: true,
+		}),
+		Entry("long duration", setRotationDeadlineCase{
+			notBefore:    -6 * 30 * 24 * time.Hour,
+			notAfter:     6 * 30 * 24 * time.Hour,
+			shouldRotate: true,
+		}),
+		Entry("short duration", setRotationDeadlineCase{
+			notBefore:    -30 * time.Second,
+			notAfter:     30 * time.Second,
+			shouldRotate: true,
+		}),
+	)
+	Context("when waitForDeadlineAndRotate is called for the first time", func() {
+		var (
+			certsDuration                = 30 * time.Second
+			manager                      *Manager
+			client                       client.Client
+			mutatingWebhookConfiguration = &admissionregistrationv1beta1.MutatingWebhookConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "fooWebhook",
+				},
+				Webhooks: []admissionregistrationv1beta1.MutatingWebhook{
+					admissionregistrationv1beta1.MutatingWebhook{
+						Name: "fooWebhook",
+						ClientConfig: admissionregistrationv1beta1.WebhookClientConfig{
+							Service: &admissionregistrationv1beta1.ServiceReference{
+								Name:      "fooWebhook",
+								Namespace: "fooWebhook",
+							},
+						},
 					},
 				},
-			},
-		},
-	}
-
-	service := &corev1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: "fooWebhook",
-			Name:      "fooWebhook",
-		},
-	}
-
-	objs := []runtime.Object{mutatingWebhookConfiguration, service}
-
-	client := fake.NewFakeClient(objs...)
-	certsDuration := time.Minute
-	manager := NewManager(client, "fooWebhook", MutatingWebhook, certsDuration)
-	manager.waitForDeadlineAndRotate()
-	//TODO Implement ErrorsHandler to take the errors that we have at
-	//     background
-
-	err = client.Get(context.TODO(), types.NamespacedName{Name: "fooWebhook"}, mutatingWebhookConfiguration)
-	if err != nil {
-		t.Fatalf("get mutatingwebhookconfiguration: (%v)", err)
-	}
+			}
 
-	clientConfig := mutatingWebhookConfiguration.Webhooks[0].ClientConfig
-	if len(clientConfig.CABundle) == 0 {
-		t.Error("CA bundle not updated")
-	}
+			service = &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "fooWebhook",
+					Name:      "fooWebhook",
+				},
+			}
+		)
+		BeforeEach(func() {
```

Comment:
can you add a check that there are no caBundles & secrets?

Reply:
We are creating a new FakeClient so there should be none I suppose, right ?

Reply:
yeah but if in the future there will be more then 1 test then we might end up with secrets from former tests.

Reply:
What I mean is that we store stuff at FakeClient memory and FakeClietn is re-created at BeforeEach so there is no issue

Reply:
ack, thanks